### PR TITLE
Add support for AES CFB mode in _CryptoExtras

### PR DESCRIPTION
**Pull request by simonjbeaumont**
## Motivation

We have AES support in `_CryptoExtras` for CBC, CTR, and GCM-SIV modes; but we do not have support for CFB.

## Modifications

- Add support for AES-CFB, backed by BoringSSL, with the same shape of API as we provide for AES CTR.
- Copy RFCVector decoder from Crypto tests to CryptoExtrasTests.
- Add test vectors from NIST.

## Result

New API added to support AES-CFB.

### CODE REVIEW

File: Tests/_CryptoExtrasTests/Utils/RFCVector.swift
Comment (simonjbeaumont) [inline Tests/_CryptoExtrasTests/Utils/RFCVector.swift:1]: Note to reviewer: This file has been copied from `CryptoTests`.

File: Sources/_CryptoExtras/AES/AES_CFB.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            nonce: AES._CFB.Nonce
+        ) throws -> Data {
+            let bytes: ContiguousBytes = ciphertext.regions.count == 1 ? ciphertext.regions.first! : Array(ciphertext)
+            return try AESCFBImpl.encryptOrDecrypt(.decrypt, bytes, using: key, nonce: nonce)
+        }
+    }
+}
+
+extension AES._CFB {
+    public struct Nonce: Sendable {
```
Comment (simonjbeaumont) [inline Sources/_CryptoExtras/AES/AES_CFB.swift:46]: FYI: Our existing AES APIs differ in calling this `IV` or `Nonce`. I aligned with CTR, which was added most recently(?).

Comment (Lukasa) [reply Sources/_CryptoExtras/AES/AES_CFB.swift:46]: For CFB nonce is acceptable, though it is probably more commonly called an IV. NIST calls it an IV, though that seems more to be out of caution than anything else.

**Review by Lukasa — APPROVED** [review]
This looks great, thanks!

File: Sources/_CryptoExtras/AES/AES_CFB.swift (hunk 1 above)
Comment (simonjbeaumont) [reply Sources/_CryptoExtras/AES/AES_CFB.swift]: Renamed `AES._CFB.Nonce` to `IV` and all `nonce` and `nonceBytes` parameters to `iv` and `ivBytes, respectively.

**Comment by simonjbeaumont** [comment]
Thanks @Lukasa I just push an update that renamed `Nonce` to `IV`, throughout. Otherwise no changes since your review. I'll leave it with you to merge whenever it's a good time.

**Pull request merged**